modules/andes-core/broker/src/main/java/org/dna/mqtt/wso2/AndesMQTTBridge.java

```diff
@@ -239,4 +242,39 @@ public void onProcessPingRequest(String clientID){
         }
         MQTTopicManager.getInstance().processPingRequest(clientID);
     }
+
+
+    /**
+     * Sends retain messages if exist upon MQTT subscription.
+     *
+     * @param topicName topic name of the subscription.
+     * @param clientID subscription client id
+     */
+    public void sendRetainMessageToSubscriber(String topicName, String clientID) {
+
+        // Send retained message if available
+        try {
+            List<DeliverableAndesMetadata> metadataList = Andes.getInstance().getRetainedMetadataByTopic(topicName);
```

Comment:
If subscriber subscribe with wildcards (#,+) it should receive retain messages for all those topics. Therefore multiple retain messages can be received per subscriber.